internal/services/loganalytics/log_analytics_workspace_resource_test.go

```diff
@@ -276,6 +276,32 @@ func TestAccLogAnalyticsWorkspace_cmkForQueryForced(t *testing.T) {
 	})
 }
 
+func TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_log_analytics_workspace", "test")
+	r := LogAnalyticsWorkspaceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withUseResourceOnlyPermission(data, false),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config: r.withUseResourceOnlyPermission(data, true),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config: r.withUseResourceOnlyPermission(data, false),
+			Check: acceptance.ComposeTestCheckFunc(
```

Comment:
```suggestion
func TestAccLogAnalyticsWorkspace_ToggleAllowOnlyResourcePermission(t *testing.T) {
	data := acceptance.BuildTestData(t, "azurerm_log_analytics_workspace", "test")
	r := LogAnalyticsWorkspaceResource{}

	data.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.withUseResourceOnlyPermission(data, true),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
		},
		{
			Config: r.withUseResourceOnlyPermission(data, false),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
			),
		},
		{
			Config: r.withUseResourceOnlyPermission(data, true),
			Check: acceptance.ComposeTestCheckFunc(
```

Reply:
Could you please teach me why it should be true -> false -> true rather than false -> true -> false? Is it related to the defaults value?

Reply:
Yes, the suggestion of true -> false -> true is under the assumption that the value is defaulted to false. When toggling a value we need to make sure that it can be set to something other than the default value, then back again.

When a new property is added to a service in Azure, older versions of the API have tended to return false for it, hence the suggestion to default the value to `false` in the schema. 

If the API is truly defaulting to true, even when creating the resource with an older API version, then false -> true -> false is the appropriate order for toggling, just make sure that we are also defaulting the correct value in the read function (it's currently defaulted to false in the read).

Reply:
Copy that, thank you. tried with a really old api version "2015-03-20", and it defaults to true, so I'm sure it does default to true. With the latest commit I'm going to resolve these two comments. thank you again.